cohort_sdk_client/README.md

```diff
@@ -1 +1,304 @@
+The role of Replicator in Talos ecosystem is to listen to certification decisions and apply transactions in the same order as they were certified.
+
+```TypeScript
+const replicator = await Replicator.init(...)
+
+const fnSnapshotProviderCallback = async () => {...}
+const fnStatemapInstallerCallback = async () => {...}
+// this will run indefinitely
+await replicator.run(fnSnapshotProviderCallback, fnTransactionInstaller)
+```
+The replicator is a server running in the background. It connects to Kafka and recieves all messages from certification topic. Regardless of the order messages were received, the replicator will re-order them and feed into your `fnStatemapInstallerCallback` in the correct order. Your role here is to implement both callbacks.
+
+## About "Snapshot Provider Callback"
+The callback has the shape of simple getter function which returns the "global state" - the latest version number which you have in your DB. The signature of the function looks like this:
+
+```TypeScript
+async (): Promise<number>
+```
+- It receives no parameters.
+- It returns the snapshot version as number
+
+This callback is invoked only once during startup of Replicator server. It fetches current snapshot. Replicator uses current snapshot version as a point in time marker which tells Replictor not to react to any messages older than that version.
+
+## About "Statemap Installer Callback"
+This statemap installer callback looks like this:
+
+```TypeScript
+async (data: JsStatemapAndSnapshot): Promise<void>
+```
+- It receives `JsStatemapAndSnapshot`
+- There is no return value, if function completes successfully SDK will proceed to handling the next transaction.
+
+```TypeScript
+export interface JsStatemapAndSnapshot {
+  statemap: Array<JsStatemapItem>
+  version: number
+}
+
+export interface JsStatemapItem {
+  action: string
+  version: number
+  safepoint?: number
+  payload: any
+}
+```
+The purpose of statemap installer callback is to "catch-up" on changes made by other cohorts (including your cohort).
+
+In the deployment where there is no Talos present, microservices typically implement eventual consistency model by utilising messaging broker. For example, Kafka. If changes are made to bank account in Cohort 1 that change will eventually propagate to Cohort 2 through messaging middleware.
+
+In the deployment with Talos, the replicator, specifically this callback, is responsible for updating the database so that your cohort has the up-to-date view of bank account (or any other shared object). The async messaging is abstracted away from you by SDK. You just provide a handler how you want your database to be updated.
+
+However, there are few rules which Talos protocols expects you to implement.
+
+- Once you are done updating business objects in your database, it is very important to update the global stapshot. The replicator, specifically this callback, is the only place within cohort which is responsible for writing into snapshot table.
```

Comment:
Small typo - 
Update 
`global stapshot` to `global snapshot`

Reply:
Lol thanks.